Deprecate ml_app; add agent_service to SDK docs (#38353)
DataDog/documentation: content/en/llm_observability/instrumentation/sdk.md

##### Parameters

`agent_service`
: optional - _string_

> <br />The name of your agent service (LLM application), under which all traces and spans are grouped. Takes precedence over `ml_app` and `service`. See [Application naming guidelines](#application-naming-guidelines) for allowed characters. To override this value for a given trace, see [Tracing multiple applications](#tracing-multiple-applications). If not provided, this defaults to the value of `DD_LLMOBS_ML_APP` or `DD_SERVICE`.

Reviewer: ```suggestion
<br />The name of your agent service (LLM application), under which all traces and spans are grouped. This value takes precedence over `ml_app` and `service`. See [Application naming guidelines](#application-naming-guidelines) for allowed characters. To override this value for a specific trace, see [Tracing multiple applications](#tracing-multiple-applications). If omitted, the value defaults to `DD_LLMOBS_ML_APP` or `DD_SERVICE`.
```

`ml_app`

> : optional - _string_ — **Deprecated.** Use `agent_service` instead. Will be removed in a future major version.

Reviewer: ```suggestion
: optional - _string_ — **Deprecated.** Use `agent_service` instead. This option will be removed in a future major version.
```

> <br />Previously: the name of your ML application. Use `agent_service` going forward.

Reviewer: ```suggestion
<br />Previously used to specify the name of your ML application.
```

: optional - _string_
<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.

`agent_service`
: optional - _string_
<br/>The agent service that this span belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information. Takes precedence over `ml_app`.

`ml_app`
: optional - _string_ — **Deprecated.** Use `agent_service` instead.

> <br/>The name of the ML application that the operation belongs to. Use `agent_service` going forward.

Reviewer: ```suggestion
<br/>The name of the ML application that the operation belongs to.
```

: optional - _string_
<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.

`agent_service`
: optional - _string_
<br/>The agent service that this span belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information. Takes precedence over `ml_app`.

`ml_app`
: optional - _string_ — **Deprecated.** Use `agent_service` instead.

> <br/>The name of the ML application that the operation belongs to. Use `agent_service` going forward.

Reviewer: ```suggestion
<br/>The name of the ML application that the operation belongs to.
```

: optional - _string_
<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.

`agent_service`
: optional - _string_
<br/>The agent service that this span belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information. Takes precedence over `ml_app`.

`ml_app`
: optional - _string_ — **Deprecated.** Use `agent_service` instead.

> <br/>The name of the ML application that the operation belongs to. Use `agent_service` going forward.

Reviewer: ```suggestion
<br/>The name of the ML application that the operation belongs to.
```

: optional - _string_
<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.

`agent_service`
: optional - _string_
<br/>The agent service that this span belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information. Takes precedence over `ml_app`.

`ml_app`
: optional - _string_ — **Deprecated.** Use `agent_service` instead.

> <br/>The name of the ML application that the operation belongs to. Use `agent_service` going forward.

Reviewer: ```suggestion
<br/>The name of the ML application that the operation belongs to.
```

: optional - _string_
<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.

`agent_service`
: optional - _string_
<br/>The agent service that this span belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information. Takes precedence over `ml_app`.

`ml_app`
: optional - _string_ — **Deprecated.** Use `agent_service` instead.

> <br/>The name of the ML application that the operation belongs to. Use `agent_service` going forward.

Reviewer: ```suggestion
<br/>The name of the ML application that the operation belongs to.
```

: optional - _string_
<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.

`agent_service`
: optional - _string_
<br/>The agent service that this span belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information. Takes precedence over `ml_app`.

`ml_app`
: optional - _string_ — **Deprecated.** Use `agent_service` instead.

> <br/>The name of the ML application that the operation belongs to. Use `agent_service` going forward.

Reviewer: ```suggestion
<br/>The name of the ML application that the operation belongs to.
```

: optional - _string_
<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.

`agent_service`
: optional - _string_
<br/>The agent service that this span belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information. Takes precedence over `ml_app`.

`ml_app`
: optional - _string_ — **Deprecated.** Use `agent_service` instead.

> <br/>The name of the ML application that the operation belongs to. Use `agent_service` going forward.

Reviewer: ```suggestion
<br/>The name of the ML application that the operation belongs to.
```

The SDK supports tracing multiple LLM applications from the same service.

> You can configure the `agent_service` argument on `LLMObs.enable()`, or use the `DD_SERVICE` environment variable, to set the agent service name for all generated spans by default.

Reviewer: ```suggestion
You can configure the `agent_service` argument on `LLMObs.enable()` or set the `DD_SERVICE` environment variable to define the default agent service name for all generated spans.
```
